plugins/digitalocean/resoto_plugin_digitalocean/resources.py

```diff
@@ -239,31 +243,153 @@ class DigitalOceanFloatingIP(DigitalOceanResource, BaseIPAddress):
 
     kind: ClassVar[str] = "digitalocean_floating_ip"
 
-    locked: bool = field(default=False)
+    is_locked: bool = field(default=False)
 
     def delete(self, graph: Graph) -> bool:
         return NotImplemented
 
 
 @dataclass(eq=False)
-class DigitalOceanCustomResource(DigitalOceanResource, BaseResource):
-    """An digitalocean custom resource that only inherits the collectors
-    DigitalOceanResource class as well as the BaseResource base class.
+class DigitalOceanImage(DigitalOceanResource, BaseResource):
+    """DigitalOcean image"""
+
+    kind: ClassVar[str] = "digitalocean_image"
+
+    do_image_distribution: str = field(default="")
+    do_image_slug: str = field(default="")
+    do_image_public: bool = field(default=False)
+    do_image_min_disk_size: int = field(default=0)
+    do_image_type: str = field(default="")
+    do_image_size_gigabytes: int = field(default=0)
+    do_image_description: str = field(default="")
+    do_image_status: str = field(default="")
+
+
+@dataclass(eq=False)
+class DigitalOceanSpace(DigitalOceanResource, BaseBucket):
+    """DigitalOcean space"""
+
+    kind: ClassVar[str] = "digitalocean_space"
+
+
+@dataclass(eq=False)
+class DigitalOceanApp(DigitalOceanResource, BaseResource):
+    """DigitalOcean app"""
+
+    kind: ClassVar[str] = "digitalocean_app"
+
+    do_app_tier_slug: str = field(default="")
+    do_app_default_ingress: Optional[str] = None
+    do_app_live_url: Optional[str] = None
+    do_app_live_url_base: Optional[str] = None
+    do_app_live_domain: Optional[str] = None
+
+
+@dataclass(eq=False)
+class DigitalOceanCdnEndpoint(DigitalOceanResource, BaseEndpoint):
+    """DigitalOcean CDN endpoint"""
+
+    kind = "digitalocean_cdn_endpoint"
+
+    do_cdn_origin: Optional[str] = None
+    do_cdn_endpoint: Optional[str] = None
+    do_cdn_created_at: Optional[str] = None
+    do_cdn_certificate_id: Optional[str] = None
+    do_cdn_custom_domain: Optional[str] = None
+    do_cdn_ttl: Optional[int] = None
+
+
+@dataclass(eq=False)
+class DigitalOceanCertificate(DigitalOceanResource, BaseCertificate):
+    """DigitalOcean certificate"""
+
+    kind = "digitalocean_certificate"
+
+    do_cert_sha1_fingerprint: Optional[str] = None
```

Comment:
Every cert has a fingerprint, lets put this into BaseCertificate as `sha1_fingerprint`